File: java2csharp/java/Bixolon/UPOS Compliant/BXLTest/POSPrinterFirmwareFragment.java
package com.bxl.postest;

import jpos.JposException;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;


public class POSPrinterFirmwareFragment extends POSPrinterFragment
		implements View.OnClickListener, FileListDialogFragment.OnClickListener {
	
	private EditText fileNameEditText;
	
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.fragment_pos_printer_firmware, container, false);
		
		fileNameEditText = (EditText) view.findViewById(R.id.editTextFileName);
		
		view.findViewById(R.id.buttonBrowse).setOnClickListener(this);
		view.findViewById(R.id.buttonGo).setOnClickListener(this);
		
		deviceMessagesTextView = (TextView) view.findViewById(R.id.textViewDeviceMessages);
		
		return view;
	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.buttonBrowse:
			FileListDialogFragment.showDialog(getFragmentManager());
			break;
			
		case R.id.buttonGo:
			try {
				posPrinter.updateFirmware(fileNameEditText.getText().toString());
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			break;
		}
	}
	
	public void onClick(String text) {
		fileNameEditText.setText(text);
	}
}

File: java2csharp/csharp/Bixolon/UPOS Compliant/BXLTest/POSPrinterFirmwareFragment.cs
namespace com.bxl.postest
{

	using JposException = jpos.JposException;
	using Bundle = android.os.Bundle;
	using LayoutInflater = android.view.LayoutInflater;
	using View = android.view.View;
	using ViewGroup = android.view.ViewGroup;
	using EditText = android.widget.EditText;
	using TextView = android.widget.TextView;


	public class POSPrinterFirmwareFragment : POSPrinterFragment, View.OnClickListener, FileListDialogFragment.OnClickListener
	{

		private EditText fileNameEditText;

		public override View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			View view = inflater.inflate(R.layout.fragment_pos_printer_firmware, container, false);

			fileNameEditText = (EditText) view.findViewById(R.id.editTextFileName);

			view.findViewById(R.id.buttonBrowse).OnClickListener = this;
			view.findViewById(R.id.buttonGo).OnClickListener = this;

			deviceMessagesTextView = (TextView) view.findViewById(R.id.textViewDeviceMessages);

			return view;
		}

		public virtual void onClick(View v)
		{
			switch (v.Id)
			{
			case R.id.buttonBrowse:
				FileListDialogFragment.showDialog(FragmentManager);
				break;

			case R.id.buttonGo:
				try
				{
					posPrinter.updateFirmware(fileNameEditText.Text.ToString());
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}
				break;
			}
		}

		public virtual void onClick(string text)
		{
			fileNameEditText.Text = text;
		}
	}

}